### PR TITLE
feat: move repo_checks here

### PR DESCRIPTION
**Pull request by kdmccormick**
### Description

* Brings in repo_checks and its tests over from https://github.com/openedx/terraform-github/tree/main/migrate
* Adds back Python 3.8 compatibility, as terraform-github was using Python 3.9. Pylint made this very easy.
* Adds a [new README](https://github.com/openedx/repo-tools/tree/kdmccormick/repo-checks/edx_repo_tools/repo_checks) with usage instructions.
* Fixes repo_checks tests, which were broken.
* Fixes repo_checks linting, which was broken (although repo-tools doesn't enforce linting and has hundreds of violations, whatever)

### Dry-run output

https://gist.github.com/kdmccormick/16ec2fde2ff4c587ada33c90ac9ce16e

Notes:
* The script outputs in reverse-chronological order
* Before  `tutor-contrib-ralph`, some checks need fixing (new repos that we haven't added the CLA check to). I can apply the fixes after merging this PR.
* Starting with `tutor-contrib-ralph`, it seems like all checks are passing
* This run ended 90% of the way through the repo list with a rate limit. I feel confident that the last 10% repos wouldn't have changed anything, but I can finish the run this afternoon if anyone wants me to.

### Supporting info

Closes https://github.com/openedx/terraform-github/issues/57

### CODE REVIEW

**Comment by kdmccormick** [comment]
Just an FYI @nedbat , since I see your name all over repo-tools :)

File: setup.py [hunk 1]
```diff
@@ -55,6 +55,15 @@ def is_requirement(line):
     slug = fextra.split("/")[1]
     EXTRAS_REQUIRE[slug] = load_requirements(fextra)
 
+# To run tests & linting across the entire repo, we need to install the union
+# of *all* extra requirements lists *plus* the dev-specific requirements.
+# If this list contains conflicting pins, then installing it will fail;
+# that is intentional. 
+EXTRAS_REQUIRE["dev"] = sorted({
+    *load_requirements("dev-requirements.txt"),
```
Comment (feanil) [inline setup.py:63]: It looks like we want to [move away from using this file](https://github.com/openedx/repo-tools/blob/master/dev-requirements.txt), so I think we should just reference `development.txt` directly.
```suggestion
    *load_requirements("requirements/development.txt"),
```

**Review by feanil — APPROVED** [review]
One note but otherwise looks good to me!

File: setup.py (hunk 1 above)
Comment (kdmccormick) [reply setup.py:63]: Good point 👍🏻

**Pull request merged**